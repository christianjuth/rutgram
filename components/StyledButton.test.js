import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Button from './StyledButton';

configure({ adapter: new Adapter() });

describe('<StyledButton />', () => {

  it('blue button renders correctly', () => {
    const tree = renderer
      .create(<Button>Blue Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('outline button renders correctly', () => {
    const tree = renderer
      .create(<Button mode='outlined'>Outline Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onPress fires correctly', () => {
    const mockFunc = jest.fn();
    const component = shallow(<Button onPress={mockFunc}>Outline Button</Button>);
    component.simulate('press');
    expect(mockFunc).toHaveBeenCalledTimes(1);
  });
});
