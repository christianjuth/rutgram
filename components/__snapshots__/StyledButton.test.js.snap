// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> blue button renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3798f0",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#ffffff",
                "fontFamily": "HelveticaNeue-Medium",
              },
              Object {
                "fontFamily": "HelveticaNeue-Medium",
              },
            ],
          ]
        }
      >
        Blue Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`<StyledButton /> outline button renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#000",
                "fontFamily": "HelveticaNeue-Medium",
              },
              Object {
                "fontFamily": "HelveticaNeue-Medium",
              },
            ],
          ]
        }
      >
        Outline Button
      </Text>
    </View>
  </View>
</View>
`;
